### ISSUE TITLE
constant NullPointerExceptions

### ISSUE THREAD

**Issue by patterkyle** [cursive-ide/cursive#1216]
When running cursive, I get constant messages at the bottom of the screen reporting a NullPointerException. (A new one pops up every minute or so.) As far as I can tell, it seems to work fine other than this. Here's the stack trace:

```
java.lang.NullPointerException
    at cursive.psi$project.invoke(psi.clj:104)
    at cursive.psi$cached_value.doInvoke(psi.clj:493)
    at clojure.lang.RestFn.invoke(RestFn.java:467)
    at cursive.types$element_type.invoke(types.clj:224)
    at cursive.intellij.method$matching_indexes$fn__4525.invoke(method.clj:65)
    at clojure.core$mapv$fn__6727.invoke(core.clj:6616)
    at clojure.lang.PersistentVector.reduce(PersistentVector.java:333)
    at clojure.core$reduce.invoke(core.clj:6518)
    at clojure.core$mapv.invoke(core.clj:6616)
    at cursive.intellij.method$matching_indexes.invoke(method.clj:65)
    at cursive.extensions.clojure.core.resolve$nary_inline$fn__6819$fn__6820.invoke(resolve.clj:605)
    at clojure.core.protocols$fn__6523.invoke(protocols.clj:167)
    at clojure.core.protocols$fn__6478$G__6473__6487.invoke(protocols.clj:19)
    at clojure.core.protocols$seq_reduce.invoke(protocols.clj:27)
    at clojure.core.protocols$fn__6506.invoke(protocols.clj:100)
    at clojure.core.protocols$fn__6452$G__6447__6465.invoke(protocols.clj:13)
    at clojure.core$reduce.invoke(core.clj:6515)
    at cursive.extensions.clojure.core.resolve$nary_inline$fn__6819.invoke(resolve.clj:603)
    at cursive.types$invocation_type$fn__4440.invoke(types.clj:137)
    at clojure.core$some.invoke(core.clj:2570)
    at cursive.types$invocation_type.invoke(types.clj:137)
    at cursive.types$calculate_element_type.invoke(types.clj:209)
    at cursive.psi$cached_value$reify__12657.compute(psi.clj:497)
    at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
    at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
    at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:66)
    at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
    at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:84)
    at cursive.psi$cached_value.doInvoke(psi.clj:500)
    at clojure.lang.RestFn.invoke(RestFn.java:467)
    at cursive.types$element_type.invoke(types.clj:224)
    at clojure.core$map$fn__4553.invoke(core.clj:2624)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:49)
    at clojure.lang.RT.seq(RT.java:507)
    at clojure.core$seq__4128.invoke(core.clj:137)
    at clojure.core.protocols$seq_reduce.invoke(protocols.clj:26)
    at clojure.core.protocols$fn__6506.invoke(protocols.clj:100)
    at clojure.core.protocols$fn__6452$G__6447__6465.invoke(protocols.clj:13)
    at clojure.core$reduce.invoke(core.clj:6515)
    at cursive.extensions.clojure.core.resolve$nary_inline$fn__6819.invoke(resolve.clj:603)
    at cursive.types$invocation_type$fn__4440.invoke(types.clj:137)
    at clojure.core$some.invoke(core.clj:2570)
    at cursive.types$invocation_type.invoke(types.clj:137)
    at cursive.types$calculate_element_type.invoke(types.clj:209)
    at cursive.psi$cached_value$reify__12657.compute(psi.clj:497)
    at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
    at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
    at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:66)
    at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
    at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:84)
    at cursive.psi$cached_value.doInvoke(psi.clj:500)
    at clojure.lang.RestFn.invoke(RestFn.java:467)
    at cursive.types$element_type.invoke(types.clj:224)
    at cursive.intellij.method$matching_indexes$fn__4525.invoke(method.clj:65)
    at clojure.core$mapv$fn__6727.invoke(core.clj:6616)
    at clojure.core.protocols$fn__6523.invoke(protocols.clj:167)
    at clojure.core.protocols$fn__6478$G__6473__6487.invoke(protocols.clj:19)
    at clojure.core.protocols$seq_reduce.invoke(protocols.clj:31)
    at clojure.core.protocols$fn__6506.invoke(protocols.clj:101)
    at clojure.core.protocols$fn__6452$G__6447__6465.invoke(protocols.clj:13)
    at clojure.core$reduce.invoke(core.clj:6519)
    at clojure.core$mapv.invoke(core.clj:6616)
    at cursive.intellij.method$matching_indexes.invoke(method.clj:65)
    at cursive.resolve.symbol.editor.clj$find_static_method.invoke(clj.clj:233)
    at cursive.resolve.symbol.editor.clj$editor_resolver$reify__5453$fn__5492.invoke(clj.clj:465)
    at clojure.core$map$fn__4553.invoke(core.clj:2624)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:49)
    at clojure.lang.RT.seq(RT.java:507)
    at clojure.core$seq__4128.invoke(core.clj:137)
    at clojure.core$apply.invoke(core.clj:630)
    at clojure.core$mapcat.doInvoke(core.clj:2660)
    at clojure.lang.RestFn.invoke(RestFn.java:423)
    at cursive.resolve.symbol.editor.clj$editor_resolver$reify__5453.resolve_qualified(clj.clj:460)
    at cursive.resolve.symbol$resolve_normal_symbol.invoke(symbol.clj:60)
    at cursive.resolve.symbol$resolve_symbol.invoke(symbol.clj:100)
    at clojure.lang.Var.invoke(Var.java:383)
    at cursive.psi.impl.symbols.SymbolReference$SymbolResolver.resolve(SymbolReference.java:67)
    at cursive.psi.impl.symbols.SymbolReference$SymbolResolver.resolve(SymbolReference.java:64)
    at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:144)
    at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:167)
    at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:159)
    at cursive.psi.impl.symbols.SymbolReference.multiResolve(SymbolReference.java:58)
    at cursive.resolve$resolve.invoke(resolve.clj:55)
    at cursive.annotator$annotate_symbol.invoke(annotator.clj:341)
    at cursive.annotator$fn__6040.invoke(annotator.clj:384)
    at cursive.annotator$fn__6025$G__6020__6032.invoke(annotator.clj:375)
    at cursive.annotator.ClojureAnnotator.annotate(annotator.clj:399)
    at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.runAnnotators(DefaultHighlightVisitor.java:162)
    at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:103)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:354)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.access$100(GeneralHighlightingPass.java:65)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass$3.run(GeneralHighlightingPass.java:288)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:308)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.access$200(GeneralHighlightingPass.java:65)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass$4.run(GeneralHighlightingPass.java:314)
    at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:87)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:311)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:280)
    at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:219)
    at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:82)
    at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:67)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1$1.run(PassExecutorService.java:444)
    at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1169)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1.run(PassExecutorService.java:435)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:446)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:432)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:408)
    at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:206)
    at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
    at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
    at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
    at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
    at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:108)



java.lang.NullPointerException
    at cursive.psi$project.invoke(psi.clj:104)
    at cursive.psi$cached_value.doInvoke(psi.clj:493)
    at clojure.lang.RestFn.invoke(RestFn.java:467)
    at cursive.types$element_type.invoke(types.clj:224)
    at cursive.intellij.method$matching_indexes$fn__4525.invoke(method.clj:65)
    at clojure.core$mapv$fn__6727.invoke(core.clj:6616)
    at clojure.lang.PersistentVector.reduce(PersistentVector.java:333)
    at clojure.core$reduce.invoke(core.clj:6518)
    at clojure.core$mapv.invoke(core.clj:6616)
    at cursive.intellij.method$matching_indexes.invoke(method.clj:65)
    at cursive.extensions.clojure.core.resolve$nary_inline$fn__6819$fn__6820.invoke(resolve.clj:605)
    at clojure.core.protocols$fn__6523.invoke(protocols.clj:167)
    at clojure.core.protocols$fn__6478$G__6473__6487.invoke(protocols.clj:19)
    at clojure.core.protocols$seq_reduce.invoke(protocols.clj:27)
    at clojure.core.protocols$fn__6506.invoke(protocols.clj:100)
    at clojure.core.protocols$fn__6452$G__6447__6465.invoke(protocols.clj:13)
    at clojure.core$reduce.invoke(core.clj:6515)
    at cursive.extensions.clojure.core.resolve$nary_inline$fn__6819.invoke(resolve.clj:603)
    at cursive.types$invocation_type$fn__4440.invoke(types.clj:137)
    at clojure.core$some.invoke(core.clj:2570)
    at cursive.types$invocation_type.invoke(types.clj:137)
    at cursive.types$calculate_element_type.invoke(types.clj:209)
    at cursive.psi$cached_value$reify__12657.compute(psi.clj:497)
    at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
    at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
    at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:66)
    at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
    at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:84)
    at cursive.psi$cached_value.doInvoke(psi.clj:500)
    at clojure.lang.RestFn.invoke(RestFn.java:467)
    at cursive.types$element_type.invoke(types.clj:224)
    at clojure.core$map$fn__4553.invoke(core.clj:2624)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:49)
    at clojure.lang.RT.seq(RT.java:507)
    at clojure.core$seq__4128.invoke(core.clj:137)
    at clojure.core.protocols$seq_reduce.invoke(protocols.clj:26)
    at clojure.core.protocols$fn__6506.invoke(protocols.clj:100)
    at clojure.core.protocols$fn__6452$G__6447__6465.invoke(protocols.clj:13)
    at clojure.core$reduce.invoke(core.clj:6515)
    at cursive.extensions.clojure.core.resolve$nary_inline$fn__6819.invoke(resolve.clj:603)
    at cursive.types$invocation_type$fn__4440.invoke(types.clj:137)
    at clojure.core$some.invoke(core.clj:2570)
    at cursive.types$invocation_type.invoke(types.clj:137)
    at cursive.types$calculate_element_type.invoke(types.clj:209)
    at cursive.psi$cached_value$reify__12657.compute(psi.clj:497)
    at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
    at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
    at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:66)
    at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
    at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:84)
    at cursive.psi$cached_value.doInvoke(psi.clj:500)
    at clojure.lang.RestFn.invoke(RestFn.java:467)
    at cursive.types$element_type.invoke(types.clj:224)
    at cursive.intellij.method$matching_indexes$fn__4525.invoke(method.clj:65)
    at clojure.core$mapv$fn__6727.invoke(core.clj:6616)
    at clojure.core.protocols$fn__6523.invoke(protocols.clj:167)
    at clojure.core.protocols$fn__6478$G__6473__6487.invoke(protocols.clj:19)
    at clojure.core.protocols$seq_reduce.invoke(protocols.clj:31)
    at clojure.core.protocols$fn__6506.invoke(protocols.clj:101)
    at clojure.core.protocols$fn__6452$G__6447__6465.invoke(protocols.clj:13)
    at clojure.core$reduce.invoke(core.clj:6519)
    at clojure.core$mapv.invoke(core.clj:6616)
    at cursive.intellij.method$matching_indexes.invoke(method.clj:65)
    at cursive.resolve.symbol.editor.clj$find_static_method.invoke(clj.clj:233)
    at cursive.resolve.symbol.editor.clj$editor_resolver$reify__5453$fn__5492.invoke(clj.clj:465)
    at clojure.core$map$fn__4553.invoke(core.clj:2624)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:49)
    at clojure.lang.RT.seq(RT.java:507)
    at clojure.core$seq__4128.invoke(core.clj:137)
    at clojure.core$apply.invoke(core.clj:630)
    at clojure.core$mapcat.doInvoke(core.clj:2660)
    at clojure.lang.RestFn.invoke(RestFn.java:423)
    at cursive.resolve.symbol.editor.clj$editor_resolver$reify__5453.resolve_qualified(clj.clj:460)
    at cursive.resolve.symbol$resolve_normal_symbol.invoke(symbol.clj:60)
    at cursive.resolve.symbol$resolve_symbol.invoke(symbol.clj:100)
    at clojure.lang.Var.invoke(Var.java:383)
    at cursive.psi.impl.symbols.SymbolReference$SymbolResolver.resolve(SymbolReference.java:67)
    at cursive.psi.impl.symbols.SymbolReference$SymbolResolver.resolve(SymbolReference.java:64)
    at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:144)
    at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:167)
    at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:159)
    at cursive.psi.impl.symbols.SymbolReference.multiResolve(SymbolReference.java:58)
    at cursive.resolve$resolve.invoke(resolve.clj:55)
    at clojure.core$map$fn__4553.invoke(core.clj:2624)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:49)
    at clojure.lang.Cons.next(Cons.java:39)
    at clojure.lang.RT.next(RT.java:674)
    at clojure.core$next__4112.invoke(core.clj:64)
    at clojure.core$concat$cat__4217$fn__4218.invoke(core.clj:707)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:56)
    at clojure.lang.RT.seq(RT.java:507)
    at clojure.core$seq__4128.invoke(core.clj:137)
    at clojure.core$map$fn__4553.invoke(core.clj:2616)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:49)
    at clojure.lang.RT.seq(RT.java:507)
    at clojure.core$seq__4128.invoke(core.clj:137)
    at clojure.core$map$fn__4553.invoke(core.clj:2616)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:49)
    at clojure.lang.RT.seq(RT.java:507)
    at clojure.core$seq__4128.invoke(core.clj:137)
    at clojure.core$filter$fn__4580.invoke(core.clj:2679)
    at clojure.lang.LazySeq.sval(LazySeq.java:40)
    at clojure.lang.LazySeq.seq(LazySeq.java:56)
    at clojure.lang.Cons.next(Cons.java:39)
    at clojure.lang.RT.next(RT.java:674)
    at clojure.core$next__4112.invoke(core.clj:64)
    at clojure.core$reduce1.invoke(core.clj:909)
    at clojure.core$set.invoke(core.clj:3960)
    at cursive.inspections.unused$unused_import.invoke(unused.clj:124)
    at clojure.lang.Var.invoke(Var.java:388)
    at cursive.api.DelayedFn.invoke(DelayedFn.java:41)
    at cursive.inspections.ClInspectionVisitor.visitElement(ClInspectionVisitor.java:32)
    at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
    at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:81)
    at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:69)
    at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.runToolOnElements(LocalInspectionsPass.java:294)
    at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.access$200(LocalInspectionsPass.java:73)
    at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$2.process(LocalInspectionsPass.java:261)
    at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$2.process(LocalInspectionsPass.java:256)
    at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:122)
    at com.intellij.concurrency.ApplierCompleter.access$000(ApplierCompleter.java:44)
    at com.intellij.concurrency.ApplierCompleter$1.run(ApplierCompleter.java:85)
    at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1169)
    at com.intellij.concurrency.ApplierCompleter$2.run(ApplierCompleter.java:94)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:446)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:443)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
    at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:106)
    at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:82)
    at jsr166e.CountedCompleter.exec(CountedCompleter.java:684)
    at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
    at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
    at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
    at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
    at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:108)

```

**Comment by MerelyAPseudonym** [comment]
I've noticed this happening too, but it seems to only occur in certain projects. I'll subscribe to this issue in case @cursive-ide wants to work out what the problematic project configuration is.

**Comment by yatesco** [comment]
I see this too - I assumed my code was just too bad :-).

**Comment by DirkArnoldt** [comment]
I have this issue, too. It occurs when I open a specific source file. Other source files work well.
The source file in question has the following namespace declaration

```
(ns foo.bar
  (:require foo.core)
  (:import (foo.core SomeProtocol)))
```

and defines a record that implements the imported protocol `SomeProtocol`.

**Comment by cursive-ide** [comment]
This is fixed in the latest EAP as #1560.
